numpyro/distributions/transforms.py

```diff
@@ -790,6 +791,72 @@ def log_abs_det_jacobian(self, x, y, intermediates=None):
         x_abs = jnp.abs(x)
         return -x_abs - 2 * jnp.log1p(jnp.exp(-x_abs))
 
+class Simplex2OrderedTransform(Transform):
+    """
+    Transform a simplex into an ordered vector (via difference in Logistic CDF between cutpoints)
+    Used in [1] to induce a prior on latent cutpoints via transforming ordered category probabilities.
+    Anchor point is a nuisance parameter to improve identifiability.
+     
+    **References:**
+
+    1. *Ordinal Regression Case Study, section 2.2*,
+       M. Betancourt, https://betanalpha.github.io/assets/case_studies/ordinal_regression.html
+    """    
+    domain = constraints.simplex
+    codomain = constraints.ordered_vector
+    
+    def __init__(self, anchor_point=0.0):
```

Comment:
Could you add docs to this anchor_point? I guess we assume it is scalar for simplicity?

Reply:
I've added a dedicated one-liner instead of the brief mention that was there before. Is that what you had in mind?

":param anchor_point: Anchor point is a nuisance parameter to improve the identifiability of the transform. For simplicity, we assume it is a scalar value."

Reply:
I think you can link it to the reference. As a user, it is still not clear for me how to interpret it. If we allow non-scalar values, what is the expected shape? Should it be the same as `x`, or broadcastable to `x`,... It also depends on your implementation. Based on the implementation, I guess `anchor_point.shape` should be broadcastable to `x.shape[:-1]`?

Reply:
So my interpretation is that it's like a "baseline affinity" for the map between proba and cutpoints to be one-to-one. Practically, we could exclude it (because it would be implicitly 0.0, ie, centered around 0.0 anchor point), but we need to make sure there is no intercept in your model (or a fixed covariate). Otherwise, we would introduce the non-identifiability again. I think it's just MB being very precise and explicit rather than implicit.

***
I think you're right with the shape `x.shape[:-1]` - it absolutely must be fixed over the event_shape (axis=-1). 
However, if we are modelling cutpoints that should come from the same distribution (same `true` model, just batched) then it must be fixed (assuming it's the "baseline affinity" that effectively calibrates/anchors the cutpoints). 
Ie, I don't see the need to support anything but scalar - WDYT?

Reply:
Thanks for the clear explanation! Now, I understand what anchor_point is and see why you wanted to define `OrderedToSimplexTransform` (because `anchor_point` is more related to cut_points than the probability). Would it make more sense to construct it instead of `SimplexToOrderedTransform`?

Regarding the shape of `anchor_point`, I just meant to let users know what's its expected shape. I think you can either mentioned that `anchor_point` shape should be broadcastable to `x.shape[:-1]` or add some assertation in the call method.

Reply:
I've added the comment into docs. As for OrderedToSimplex - I'd prefer to stick to the current implementation, as it feels more natural to transform Dirichlet in call() direction.

Reply:
Thanks, yep let's stick to OrderedToSimplex. :)